src/metabase/automagic_dashboards/core.clj

```diff
@@ -431,13 +436,14 @@
    If there are multiple FKs pointing to the same table, multiple entries will
    be returned."
   [table]
-  (->> (db/select Field
-         :table_id           (:id table)
-         :fk_target_field_id [:not= nil])
-       field/with-targets
-       (filter #(some-> % :target mi/can-read?))
-       (map (fn [{:keys [id target]}]
-              (-> target field/table (assoc :link id))))))
+  (when (instance? (type Table) table)
+    (->> (db/select Field
+                    :table_id           (:id table)
+                    :fk_target_field_id [:not= nil])
+         field/with-targets
+         (filter #(some-> % :target mi/can-read?))
+         (map (fn [{:keys [id target]}]
+                (-> target field/table (assoc :link id)))))))
```

Comment:
This might be more readable like 

```clj
(for [{:keys [target id], :as field} (field/with-targets (db/select Field ...))
      :when (and target (mi/can-read? target))]
   (assoc (field/table target)
     :link id))
```